### PR TITLE
[2019-02][ci] Use a shell script to reset HEAD for submodules

### PR DESCRIPTION
**Pull request by lambdageek**
Work around a weird bug in Homebrew git 2.22.0

### CODE REVIEW

File: scripts/ci/pipeline/sdks-archive.groovy [hunk 1]
```diff
@@ -80,7 +80,7 @@ def archive (product, configuration, platform, chrootname = "", chrootadditional
 
                 // remove old stuff
                 sh 'git reset --hard HEAD'
-                sh 'git submodule foreach --recursive git reset --hard HEAD'
+                sh 'git submodule foreach --recursive `pwd`/scripts/ci/git-reset-hard.sh HEAD'
```
Comment (CoffeeFlux) [inline scripts/ci/pipeline/sdks-archive.groovy]: Maybe add a note here that this should be reverted once the bug is fixed?

Comment (lambdageek) [reply scripts/ci/pipeline/sdks-archive.groovy]: well it's not gonna stop working once the bug is fixed.

and it's maybe enough of a WTF that the curious person would read the comment in `git-reset-hard.sh` if they decide to clean this up in the future.

**Comment by lambdageek** [comment]
@monojenkins backport to 2019-06
@monojenkins backport to master

**Comment by lambdageek** [comment]
@monojenkins build failed

File: scripts/ci/git-reset-hard.sh [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+#  git submodule foreach --recursive git reset --hard HEAD
+# errors out with:
+#
+# error: unknown option `hard'
+# usage: git submodule--helper foreach [--quiet] [--recursive] [--] <command>
+
+# which is nonsense because `git foreach -h` says:
+# usage:
+#    ...
+#    or: git submodule [--quiet] foreach [--recursive] <command>
```
Comment (radical) [inline scripts/ci/git-reset-hard.sh]: IMHO, it would be useful to add these comments to the commit message too.

**Comment by lambdageek** [comment]
Added a better commit message and a note in the groovy script to revert this once the Git bug is fixed.

**Comment by lambdageek** [comment]
@monojenkins backport to 2019-06
@monojenkins backport to master

**Pull request merged**